javasource/restservices/util/JsonDeserializer.java

```diff
@@ -66,101 +66,180 @@ else if (jsonValue instanceof Long || jsonValue instanceof Double || jsonValue i
+
+				IMendixObjectMember<?> otherMember = otherSideTarget.getMember(context, otherTargetattr);
+
+				//Reference for arrays
+				if (otherMember instanceof MendixObjectReference)
+				{
+					JSONArray jsonArray = object.getJSONArray(attr);
+					readJsonArrayIntoMendixObject(context, jsonArray, otherSideTarget, autoResolve, association.getName().split("\\.")[1], target.getId());
+				}
 			}
 		}
 		Core.commit(context, target);
 	}
 
+	private static void readJsonArrayIntoMendixObject(IContext context, JSONArray object, IMendixObject target, boolean autoResolve, String associationName, IMendixIdentifier mainIdentifier) throws JSONException, Exception {
+
+		Map<String, String> attributeNameMap = buildAttributeNameMap(target.getMetaObject());
+
+		String targetattr =  attributeNameMap.get(associationName.toLowerCase().replaceAll("[^a-zA-Z0-9_]","_"));
+		IMetaAssociation association =  Core.getMetaAssociation(targetattr);
+		String otherSideType = association.getParent().getName();
+
+		for(int i = 0; i < object.length(); i++)
+		{
+			IMendixObject otherSideTarget = Core.instantiate(context, otherSideType);
+			IMendixObjectMember<?> otherMember = otherSideTarget.getMember(context, targetattr);
+			((MendixObjectReference)otherMember).setValue(context, mainIdentifier);
+
+			Object childJsonValue = object.get(i);
+			if(childJsonValue instanceof JSONArray)
+			{
+				JSONArray childJsonArray = (JSONArray)childJsonValue;
+				Object[] attributeValues =attributeNameMap.keySet().toArray();
+
+				String otherMemberName = associationName;
+				for(int a = 0; a < attributeValues.length; a++)
+				{
+					String v = (String)attributeValues[a];
+					if(!v.equals(associationName))
+						otherMemberName = v;
+				}
+
+				readJsonArrayIntoMendixObject(context, childJsonArray, otherSideTarget, autoResolve, otherMemberName, target.getId());
+			}
+			else
+			{
+				readJsonDataIntoMendixObject(context, childJsonValue, otherSideTarget, autoResolve);
+			}
+
+			Core.commit(context, otherSideTarget);
+		}
+	}
+
 	private static final Map<String, Map<String,String>> metaAttributeMaps = new HashMap<String, Map<String, String>>();
-	
+
 	private static Map<String, String> buildAttributeNameMap(IMetaObject metaObject) {
 		if (metaAttributeMaps.containsKey(metaObject.getName()))
 			return metaAttributeMaps.get(metaObject.getName());
-		
+
 		Map<String, String> attrMap = new HashMap<String,String>();
-		
+
 		for(IMetaAssociation assoc : metaObject.getMetaAssociationsParent()) {
 			String name = assoc.getName().split("\\.")[1];
 			attrMap.put(name.toLowerCase(), assoc.getName());
 			if (name.startsWith("_"))
 				attrMap.put(name.substring(1).toLowerCase(), assoc.getName());
 		}
-		
+
+		for(IMetaAssociation assoc : metaObject.getMetaAssociationsChild()) {
+			String name = assoc.getName().split("\\.")[1];
+			attrMap.put(name.toLowerCase(), assoc.getName());

```

Comment:
Might be wise here to check whether that name isn't already taken? for example by an attribute or outgoing assoc (so that would be a self ref). In that case the child assoc should the least in prio for predictability purposes?